Mathlib/Algebra/EuclideanDomain/Basic.lean

```diff
@@ -318,6 +318,66 @@ theorem mul_div_mul_comm_of_dvd_dvd {a b c d : R} (hac : c ∣ a) (hbd : d ∣ b
   rw [mul_div_cancel_left₀ _ hc0, mul_div_cancel_left₀ _ hd0, mul_mul_mul_comm,
     mul_div_cancel_left₀ _ (mul_ne_zero hc0 hd0)]
 
+theorem div_add_of_dvd (x y z : R) (h1 : y ≠ 0) (h2 : y ∣ x) : x / y + z = (x + y * z) / y := by
+  apply eq_div_of_mul_eq_right h1
+  rw [mul_add, EuclideanDomain.mul_div_cancel' h1 h2]
```

Comment:
For `Int` and `Nat` this is called [`add_mul_div_left`](https://leanprover-community.github.io/mathlib4_docs/Init/Data/Nat/Div/Basic.html#Nat.add_mul_div_left) and stated in reverse. Can you use loogle to search for all of these (and their reverses, and then match the names to the Int/Nat ones?

Reply:
I did change the names of the ones I did find. I also added the `_right` versions.